### ISSUE TITLE
Generalize iter_cloned_collect to .copied()

### ISSUE THREAD

**Issue by kpreid** [rust-lang/rust-clippy#6703]
I ran Clippy on some previously unlinted code and found that it reported that 

```rust
some_slice.iter().cloned().collect::<Vec<_>>()
```

could be replaced with `some_slice.to_vec()`, but, did _not_ report cases where I had happened to write

```rust
some_slice.iter().copied().collect::<Vec<_>>()
```

instead. Perhaps it would make sense to generalize `iter_cloned_collect` to also detect uses of `.copied()`, or add a corresponding lint if they should be kept separate for some reason.

**Comment by giraffate** [comment]
Thanks for suggesting enhancement! But it seems to be a duplicate of https://github.com/rust-lang/rust-clippy/issues/3870, so I'm closing this.

**Comment by kpreid** [comment]
I think these are **different** issues: that issue talks about a lint replacing `.cloned()` with `.copied()`. In this issue, I am proposing that the lint would recommend *replacing a chain containing `.copied()`* with `.to_vec()`.

(Incidentally, I checked the standard library source and `.to_vec()` uses `Copy` when available, so this would not have any performance regression.)

**Comment by giraffate** [comment]
Oh, sorry for my misunderstanding. I reopen this.

**Comment by HarrisonMc555** [comment]
I'll try implementing this! @rustbot claim
